.github/workflows/staging_deploy.yml

```diff
@@ -0,0 +1,35 @@
+name: cd
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  staging_deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+
+      - name: Start the ssh-agent
+        uses: webfactory/ssh-agent@v0.7.0
+        with:
+          ssh-private-key: ${{ secrets.SSH_PRIVATE_KEY }}
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          aws-access-key-id    : ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region           : us-east-1
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+
+      - name: Install kamal
+        run: gem install kamal -v 1.2.0
+
+      - name: Run deploy command
+        run: kamal deploy -d staging
```

Comment:
We still have to figure out how to cache docker builds in order to speed up deploys, but we will leave that a future PR. For more info:

https://github.com/basecamp/kamal/discussions/332